api/localai/backend_monitor.go

```diff
@@ -140,3 +151,18 @@ func BackendMonitorEndpoint(bm BackendMonitor) func(c *fiber.Ctx) error {
 		return c.JSON(status)
 	}
 }
+
+func BackendShutdownEndpoint(bm BackendMonitor) func(c *fiber.Ctx) error {
+	return func(c *fiber.Ctx) error {
+
+		client, _, err := bm.getClientFromCtx(c)
+		if err != nil {
+			return err
+		}
+		status, rpcErr := client.UnloadModel(context.TODO())

```

Comment:
I would rather suggest to _not_ unload with grpc and delegating to the backend freeing the model, but rather stop the process and let the OS clean it up. 

For two reasons mainly:
- Not all backends support unloading, thus this approach would not scale much on the long run and would force to specific workarounds
- Even if unloading would work for all, I think it's better off kill the process and be sure there are no long-standing memory allocations. That's more safe as would not expose us to eventual backend bugs.

The other aspect that doesn't make me fancy this approach is that _if_ a backend becomes unresponsive due to high load, the API would not be capable to free resources - which would not work in high load situations or constrained environment (thinking at ARM, for instance). 

I'd keep Unloading if it was one process, but since we are delegating to multiple processes with gRPC already, I think it's just more simple and safe to stop the process

Reply:
After more testing of this branch on my faster machine, I'm inclined to agree with you:

- Even after we unload the llama backend, a _significant_ portion of memory remains allocated - on my machines it's in the ballpark of 4GB.
- Killing the process is simple, and will work without regard to backend implementation language
- I agree that freezing under high load would be the _worst_ time to be unable to free!


I think when we go to implement LORA / GGUF stuff, there may be a need to invoke the "Free" methods... so I may leave some of that code around for now temporarily, since I intend to pick that up soon. In the next few commits, I'll transition the `/backend/shutdown` endpoint to be process-based instead!